api/v1beta1/kafkacluster_types.go

```diff
@@ -225,20 +225,23 @@ type AlertManagerConfig struct {
 
 // ExternalListenerConfig defines the external listener config for Kafka
 type ExternalListenerConfig struct {
-	Type                 string `json:"type"`
-	Name                 string `json:"name"`
+	*CommonListenerSpec  `json:",inline"`
```

Comment:
Is `CommonListenerSpec` optional? If not than it's no worth making it a pointer.